### ISSUE TITLE
Prep for Django 5.0

### ISSUE THREAD

**Issue by jamienoss** [chrisclark/django-sql-explorer#552]
Hi, here's a heads up on a future issue that I just stumbled upon. I have no idea about the full extent but do at least know of one change in Django 5.0 that will break sql-explorer.

The passing of unsaved model instances to related filters will be fully deprecated see [https://github.com/django/django/pull/16432](https://github.com/django/django/pull/16432/files#diff-26858a889a707910cfc9495bad2161006c7642159e41346b9242d280cffc1affR45).

This breaks at least the following code in this package ([views.utils](https://github.com/groveco/django-sql-explorer/blob/master/explorer/views/utils.py#L39-L42)) when using the Playground:
```python
user = request.user
is_favorite = False
if user.is_authenticated:
  is_favorite = QueryFavorite.objects.filter(user=user, query=query).exists()
```

``QueryFavorite.objects.filter(user=user)`` is ok, however, ``QueryFavorite.objects.filter(query=query)`` isn't and raises ``ValueError("Model instances passed to related filters must be saved.")`` because ``query`` isn't saved and therefore has no ``pk``.

**Comment by marksweb** [comment]
@jamienoss thanks for raising. The project does indeed need a bit of attention to get it ready for Django 5.0 and, if I haven't already, start testing against the main branch of Django.

Unfortunately I'm struggling for time lately so it's not something I've been able to do. But please feel free to contribute if you can offer anything & raise a PR.

**Comment by marksweb** [comment]
Just looking at updating tox and adding testing against django 5.0 and django main and the restrictions on model instances is indeed a problem for the `Query` and `QueryLog` models. The `QueryLog` has this property;

```python
    @property
    def is_playground(self):
        return self.query_id is None
```

This needs a new approach becuase we need to save the `Query` object that used to render the playground. Maybe it's as simple as adding a boolen field for `is_playground` then we don't have to change logic using that property for `QueryLog`.

But this needs some thought because the playground object's don't persist at the moment. Maybe a user should only have 1 "playground" object, so we do a `get_or_create` based on user & a playground identifier.

@chrisclark I think this may be the biggest priority as it'll block django 5.0 support.

**Comment by chrisclark** [comment]
Resolved via PR #566 and specifically this change:

https://github.com/chrisclark/django-sql-explorer/pull/566/files#diff-afd42b4ac3a28f897844add70505ddfa46ecc03770c983f7231ce29f0046a09a
